csharp/src/Ice/PropertiesI.cs

```diff
@@ -153,68 +176,13 @@ public void setProperty(string key, string val)
             throw new InitializationException("Attempt to set property with empty key");
         }
 
-        //
-        // Check if the property is legal.
-        //
-        Logger logger = Util.getProcessLogger();
-        int dotPos = key.IndexOf('.');
-        if (dotPos != -1)
-        {
-            string prefix = key.Substring(0, dotPos);
-            foreach (var validProps in Ice.Internal.PropertyNames.validProps)
-            {
-                string pattern = validProps[0].pattern();
-                dotPos = pattern.IndexOf('.');
-                Debug.Assert(dotPos != -1);
-                string propPrefix = pattern.Substring(1, dotPos - 2);
-                bool mismatchCase = false;
-                string otherKey = "";
-                if (!propPrefix.ToUpper().Equals(prefix.ToUpper()))
-                {
-                    continue;
-                }
-
-                bool found = false;
-                foreach (var prop in validProps)
-                {
-                    Regex r = new Regex(prop.pattern());
-                    Match m = r.Match(key);
-                    found = m.Success;
-                    if (found)
-                    {
-                        if (prop.deprecated())
-                        {
-                            logger.warning("deprecated property: " + key);
-                            if (prop.deprecatedBy() != null)
-                            {
-                                key = prop.deprecatedBy();
-                            }
-                        }
-                        break;
-                    }
+        // Find the property, log warnings if necessary
+        var prop = findProperty(key, true);
 
-                    if (!found)
-                    {
-                        r = new Regex(prop.pattern().ToUpper());
-                        m = r.Match(key.ToUpper());
-                        if (m.Success)
-                        {
-                            found = true;
-                            mismatchCase = true;
-                            otherKey = prop.pattern().Replace("\\", "").Replace("^", "").Replace("$", "");
-                            break;
-                        }
-                    }
-                }
-                if (!found)
-                {
-                    logger.warning("unknown property: " + key);
-                }
-                else if (mismatchCase)
-                {
-                    logger.warning("unknown property: `" + key + "'; did you mean `" + otherKey + "'");
-                }
-            }
+        // If the property is deprecated by another property, use the new property key
+        if (prop != null && prop.deprecatedBy() != null)
+        {
+            key = prop.deprecatedBy();
```

Comment:
This is unrelated to your changes, but the behavior here seems odd. The code assumes the new property values are compatible with the deprecated property.

Reply:
Yes, I want to get rid of this code as well.